src/pymmcore_plus/mda/_engine.py

```diff
@@ -567,6 +580,42 @@ def _set_event_z(self, event: MDAEvent) -> None:
         correction = self._z_correction.setdefault(p_idx, 0.0)
         self._mmc.setZPosition(cast("float", event.z_pos) + correction)
 
+    def _set_event_slm_image(self, event: MDAEvent) -> None:
+        try:
```

Comment:
for reasons of typing, we either need to "recheck" the event here:

```python
if not event.slm_image:
    return
```
or cast it with `typing.cast` if we're sure this method will never be called with a null slm_image.

```python
slm_image = cast('useq.SLMImage', event.slm_image)
```

the first option is safer, and probably fine.  let's go with that

Reply:
added first option